app/bundles/ChannelBundle/Model/FrequencyActionModel.php

```diff
@@ -62,9 +62,7 @@ private function updateFrequencyRules(Lead $contact, array $params, $preferredCh
             $frequencyRule->setChannel($channel);
             $frequencyRule->setLead($contact);
 
-            if (!$frequencyRule->getDateAdded()) {
```

Comment:
@escopecz there is no point of using this check, since `DateTime` always sets `now`

Reply:
I don't see it in the entity. You are saying that this will return a `DateTime` object?
```php
$frequencyRule = new FrequencyRule();
var_dump($frequencyRule->getDateAdded());
```
I would rather not change business logic in this PR.

Reply:
According to `phpstan` this line always evaluate to `false`, maybe the whole block should be removed as it is not impacting the logic

```
if (!$frequencyRule->getDateAdded()) {
````
```
 ------ --------------------------------------------- 
  Line   FrequencyActionModel.php                     
 ------ --------------------------------------------- 
  65     Negated boolean expression is always false.  
 ------ --------------------------------------------- 
```

Reply:
That can be fixed 2 ways:
1. the way you did
2. Check that the `getDateAdded()` method has correct type hint.

I think it does not when you consider the code snippet I shared. It should be nullable.

Reply:
I fixed type hint in this commit c6ea82219dc0bb654e719c01c04ef768eb9054cc